drivers/char/rust_example.rs

```diff
@@ -101,10 +101,9 @@ impl KernelModule for RustExample {
             let cv = Pin::from(Box::try_new(unsafe { CondVar::new() })?);
             condvar_init!(cv.as_ref(), "RustExample::init::cv1");
             {
-                let guard = data.lock();
-                #[allow(clippy::while_immutable_condition)]
+                let mut guard = data.lock();
```

Comment:
Do you mean the one that was `allow`ed before?

Also: I am always ambivalent about changing code just because a linter isn't happy...

Reply:
> Do you mean the one that was `allow`ed before?

I thought clippy wasn't able to detect changes in this case because the variable is not shared, so it really can't change (i.e., clippy's warning is legitimate). However, when I added an example with a shared variable (in the next commit of this PR), clippy was also complaining about that.

> Also: I am always ambivalent about changing code just because a linter isn't happy...

It think it makes sense to change it `&mut` because we are releasing the lock, so it is very much possible for state to change. Being `&mut` makes it explicit.